app/src/androidTest/java/org/piwigo/ui/launcher/LoginTest.java

```diff
@@ -0,0 +1,126 @@
+package org.piwigo.ui.launcher;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.piwigo.R;
+import org.piwigo.ui.main.MainActivity;
+
+import androidx.test.espresso.ViewInteraction;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.LargeTest;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.pressImeActionButton;
+import static androidx.test.espresso.action.ViewActions.replaceText;
+import static androidx.test.espresso.action.ViewActions.scrollTo;
+import static androidx.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.contrib.DrawerActions.open;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withContentDescription;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.CoreMatchers.allOf;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class LoginTest {
+
+    @Rule
+    public ActivityScenarioRule<LauncherActivity> activityScenarioRule =
+        new ActivityScenarioRule<LauncherActivity>(LauncherActivity.class);
+
+    @Test
+    public void loginTest() {
+
+        try {
+            Thread.sleep(1700);
```

Comment:
1700 is an interesting magic number, why did you choose it? Is it to wait until the splash screen is gone? -> This would raise the question whether this splash screen makes sense at all. It feels a bit like mainly wasting time... Maybe it should be shown only on the very first start of the app?

Reply:
I'm actually quite suprised that sleeps are such an accepted measure in android tests. And yes, it's a magic number.

It's not a good solution, it's the number that worked for emulator and real device - but a loop is the better choice. I just wanted to present a first version before improving

Reply:
https://stackoverflow.com/questions/21417954/espresso-thread-sleep - none of the solutions presented are *cool*

Reply:
to be honest I don't see a big problem with such a sleep here in the test. We just should try to document the influences that we can imagine. Such that it is easy to identify if we change something and we need to adjust this number to fix the test.

A short comment why we need this could be perfect and much better than blowing up this test code to make it more beautiful.

Reply:
Let's just keep it as is... if the test fails, we will anyhow need to dig into it and directly see that there is a sleep. A try with a linger time will be so straight forward.